File: app/src/main/java/com/example/localai/data/network/Catalog.java
package com.example.localai.data.network;

import java.util.List;

/** 模型目录（backend catalog.schema.json 的客户端镜像）；签名由 catalog.sig 承载。 */
public final class Catalog {

    public long schemaVersion;
    public String generatedAt;
    public List<Entry> models;

    public static class Entry {
        public String modelId;
        public String version;
        public String displayName;
        public String description;
    }
}

File: app/src/main/kotlin/com/example/localai/data/network/Catalog.kt
package com.example.localai.data.network

/** 模型目录（backend catalog.schema.json 的客户端镜像）；签名由 catalog.sig 承载。 */
class Catalog {
    @JvmField var schemaVersion: Long = 0
    @JvmField var generatedAt: String? = null
    @JvmField var models: List<Entry>? = null

    class Entry {
        @JvmField var modelId: String? = null
        @JvmField var version: String? = null
        @JvmField var displayName: String? = null
        @JvmField var description: String? = null
    }
}
